### ISSUE TITLE
Join (reindexing) fails when table spans multiple pages

### ISSUE THREAD

**Issue by realratchet** [root-11/tablite#110]
Reindexing task fails when the column spans multiple pages, here's a way to reproduce it easily. This can happen when pages are highly fragmented like in this case and probably even when with larger tables spanning multiple pages.

```
from tablite import Table
from tablite.config import Config

Config.PAGE_SIZE = 1

cat_3 = Table({'order_LP_pallets': [15060, 15060],

                'avg_layers_on_order_LP_pallets': [3.2756972111553786, 3.2756972111553786],
                'scenario': ['LP_SCPT', 'LP_MCPT']})

cat_5 = Table({'order_boxes': [23552, 24441, 23552, 24431],
                'avg_HUs_per_box': [4.162746263586956,
                                    4.146884333701567,
                                    4.162746263586956,
                                    4.147558429863698],
                'order_pallets': [41398, 41438, 37950, 37991],
                'avg_case_per_order_pallet': [43.975167882506405,
                                                43.95417249867272,
                                                42.61736495388669,
                                                42.59450922586929],
                'scenario': ['SCPT', 'MCPT', 'LP_SCPT', 'LP_MCPT']})

Config.PAGE_SIZE = 1000000

join_0 = cat_5.join(
    cat_3,
    left_keys=["scenario"],
    right_keys=["scenario"],
    kind="left",
    merge_keys=True,
    left_columns=None,
    right_columns=None
)
```

**Comment by root-11** [comment]
Im on it

**Comment by root-11** [comment]
Fixed in [a3d4a0c](https://github.com/root-11/tablite/commit/a3d4a0cfb0c6c76410b140f8521eab02605a29f2)
